arduino/smartcar/smartcar.ino

```diff
@@ -77,31 +75,53 @@ void setup() {
  
   mqtt.subscribe("/smartcar/control/#", 1);
   mqtt.onMessage([](String topic, String message) {
-    if (topic == "/smartcar/control/throttle") {
-      car.setSpeed(message.toInt());
-    } else if (topic == "/smartcar/control/steering") {
-      car.setAngle(message.toInt());
+    if (topic == "/smartcar/control/takeInput") {
+      takeInput(message);
     } else {
       Serial.println(topic + " " + message);
     }
   });
 }
  
-void obstacle() {
-    const auto distance = front.getDistance();
- if (distance > 0 && distance < 100) {
-    car.setSpeed(0);
- }
-}
- 
-void carGo() {
-      car.setSpeed(60);
-}
- 
+void takeInput(String input) {
+        int inputSelection = input.substring(0).toInt();
```

Comment:
Is this even correct?
`substring(0)` means that you are basically taking the entire string.
It does **not** mean you are taking the first character of a string.

Reply:
The sketch has compiled and the app controls the vehicle. But perhaps there is an alternative and simple way to re-code this. I know this is the solution that worked for @Adrianmeda

Reply:
I actually tried with (1) at first but it didnt work. But four rows lower we make sure that its the first number being proccesed only.

Reply:
Please look at how `substring` works in `Arduino String`. The code looks wrong and if it works it's by accident.

Reply:
Needless to say one would reasonably expect that such mistakes would be caught by the code reviewers. :warning:

Reply:
Not so familiar with substring in Arduino but will look it up, thanks.